portfolio/src/main/webapp/script.js

```diff
@@ -12,19 +12,20 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
- /*
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+function getData() {
+  fetch('/data').then(response => response.json()).then((quote) => {
+    len = quote.length;
+  text = "<ul>";
+  for (i = 0; i < len; i=i+2){
+      if(quote[i] === "") {continue;}
+      text += quote[i] + " -:<br>" + quote[i+1] + "<br>" ;
```

Comment:
quote[i] + " -:<br>" + quote[i+1] + "<br>"

^ 
Use String.format here to create a single string with the desired variable interpolation. Using '+' creates a new string each time with every new addition. This causes a lot of processing.

https://stackoverflow.com/questions/6389827/string-variable-interpolation-java#6389845

Reply:
My bad use Javascript's string interpolation instead of java:
https://dmitripavlutin.com/string-interpolation-in-javascript/